tests/parser/types/test_bytes.py

```diff
@@ -333,3 +334,61 @@ def assign():
 @pytest.mark.parametrize("code,exc", cases_invalid_assignments)
 def test_invalid_assignments(get_contract, assert_compile_failed, code, exc):
     assert_compile_failed(lambda: get_contract(code), exc)
+
+cases_invalid_types = [
+    (
+        ABI_GIntM,
+        [
+            (0, False),
+            (7, False),
+            (300, True),
+            (300, False)
+        ],
+        CompilerPanic
+    ),
+    (
+        ABI_FixedMxN,
+        [
+            (0, 0, False),
+            (8, 0, False),
+            (256, 81, True),
+            (300, 80, False)
+        ],
+        CompilerPanic
+    ),
+    (
+        ABI_BytesM,
+        [
+            (0,),
+            (33,),
+            (-10,),
+        ],
+        CompilerPanic
+    ),
+    (
+        ABI_Bytes,
+        [
+            (-1,),
+            (-69,),
+        ],
+        CompilerPanic
+    ),
+    (
+        ABI_DynamicArray,
+        [
+            (ABI_GIntM(256, False), -1),
+            (ABI_String(256), -10),
+        ],
+        CompilerPanic
+    ),
+]
+
+# double parametrization cannot work because the 2nd dimension is variable
+cases_invalid_types_flat = []
+for cases in cases_invalid_types:
+    for case in cases[1]:
+        cases_invalid_types_flat.append((cases[0], case, cases[2]))
```

Comment:
i think it will be more readable to move this loop inside the unit test

Reply:
Mh, we could do that but the advantage of doing it out of the test was to really parametrize it with these variants